src/reducer.js

```diff
@@ -247,7 +247,7 @@ const onUpdateTokenHistory = (state, action) => {
       [token]: {
         ...state.tokensHistory[token],
         data: [
-          ...state.tokensHistory[token].data,
+          ...get(state.tokensHistory, `${token}.data`, []),
```

Comment:
If you do this and we download the history later, we will end up having this tx duplicated, or I am misunderstanding the events flow?